proposals/DOM-Parts-Imperative.md

```diff
@@ -256,156 +212,67 @@ had initially contained `mailto:` before `{email}` but we could not capture this
 prefix in the attribute value because `AttributePart` could only set the whole
 attribute value.
 
-There are a few options for how to support the use case of updating attributes
-with embedded static content such as `mailto:`
-
-### Option 1. Create Multiple `AttributePart`s Together
-
-In this approach, `AttributePart` gets a new static function which creates a
-list of `AttributePart`s which work together to set a value when the values are
-to be committed:
-
-```js
-const [firstName, lastName] = AttributePart.create(element, "title", null, [
-  null,
-  " ",
-  null,
-]);
-// Syntax to be improved. Here, a new AttributePart is created between each string.
-```
-
-- **Pros**: Simplicity.
-- **Cons**: Coming up with a nice syntax to create a sequence of AttributePart
-  and string can be tricky.
-
-### Option 2. Introduce `AttributePartGroup`
-
-In this approach, we group multiple `AttributePart`s together by creating an
-explicit group:
-
-```js
-const firstName = new AttributePart();
-const lastName = new AttributePart();
-const group = AttributePartGroup(element, "title");
-group.append(firstName, " ", lastName);
-```
-
-This is morally equivalent to option 1 except there is an explicit grouping
-step.
-
-- **Pros**: Nicer syntax by the virtue of individual "partial" `AttributePart`'s
-  existence at the time of grouping. Code that assigns values to `AttributePart`
-  only needs to know about `AttributePart`
-- **Cons**: More objects / complexity. `AttributePart` will have two modes.
-
-### Option 3. Introduce `AttributePartFragment`
-
-Unlike option 2, this creates `AttributePartFragment`s from `AttributePart`,
-meaning that `AttributePart` in option 3 plays the role of `AttributePartGroup`
-in option 2:
+Instead of `AttributePart` having a single string `value`, it could optionally
+take an `Array` that contains values that should be concatenated together. This
+allows updating individually parts of the attribute without needing to serialize
+the entire string.
```

Comment:
```suggestion
Instead of `AttributePart` having a single string `value`, it could instead
take an `Array` with values that will be interleaved with those from the `strings` array. 
```

If we go with the `strings` design, we might want to require that `values` be an array, to be interleaved with the `strings`. I suppose we could take a non-array value if there's only two strings, but in practice I expect template systems to just always pass through an array here.

Reply:
I think we need to think more about the DX of this.